testserver/version/version.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package version
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Version represents a semantic version; see
+// https://semver.org/spec/v2.0.0.html.
+type Version struct {
+	major      int32
+	minor      int32
+	patch      int32
+	preRelease string
+	metadata   string
+}
+
+// Major returns the major (first) version number.
+func (v *Version) Major() int {
+	return int(v.major)
+}
+
+// Minor returns the minor (second) version number.
+func (v *Version) Minor() int {
+	return int(v.minor)
+}
+
+// Patch returns the patch (third) version number.
+func (v *Version) Patch() int {
+	return int(v.patch)
+}
+
+// PreRelease returns the pre-release version (if present).
+func (v *Version) PreRelease() string {
+	return v.preRelease
+}
+
+// Metadata returns the metadata (if present).
+func (v *Version) Metadata() string {
+	return v.metadata
+}
+
+// String returns the string representation, in the format:
+//   "v1.2.3-beta+md"
+func (v Version) String() string {
+	var b bytes.Buffer
+	fmt.Fprintf(&b, "v%d.%d.%d", v.major, v.minor, v.patch)
+	if v.preRelease != "" {
+		fmt.Fprintf(&b, "-%s", v.preRelease)
+	}
+	if v.metadata != "" {
+		fmt.Fprintf(&b, "+%s", v.metadata)
+	}
+	return b.String()
+}
+
+// versionRE is the regexp that is used to verify that a version string is
+// of the form "vMAJOR.MINOR.PATCH[-PRERELEASE][+METADATA]". This
+// conforms to https://semver.org/spec/v2.0.0.html
```

Comment:
nit: missing period.

Reply:
I'll keep it this way so that the link keeps working.